tests/pkcli/argh_test.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+"""pytest for `pykern.pkcli.argh_test`
+
+:copyright: Copyright (c) 2023 RadiaSoft, LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+
+
+def test_kwarg_to_positional():
+    from pykern.pkunit import pkre
+    from pykern.pkcollections import PKDict
+    import subprocess
+
+    p = "p"
+    k = "k"
+    pkre(
+        str(PKDict(positional_arg=p, keyword_arg=k)),
+        subprocess.check_output(
+            ("pykern", "argh_test", "kwarg-to-positional", p, "--keyword-arg", k),
```

Comment:
This will not test the right pykern. It will test pykern that's installed. This needs to test the pykern in the subdirectories. It's tricky to do this. Rather, you can call pkcli directly: `pkcli.main("pykern")`. You setup argv. Look at other pkcli tests.

Reply:
python-ci-run installs, but that's not classical CI behavior. Typically, you test before installing, e.g. make test, make install, not make install, make test. 

[Modify sys.path in the test](https://github.com/radiasoft/pykern/blob/d279e4ddccdf48165ddc1886f15ef2521b211efc/tests/pkcli_test.py#L122) so that it can find the pkcli modules.

Reply:
I see `pykern ci run` enforces the cwd is the top-level of the package under test so it is added to PYTHONPATH implicitly. 

Still if we were to run `pykern test path/to/test` which doesn't have this enforcement then I think we still test the installed version and not whatever version we are running the test file from.

I tried this by installing pykern from master. Then cloning pykern again, checking out this branch, chaning the test to have `import pykern.pkcli.argh_test`, and invoking `pykern test /path/to/pykern`. It failed with a ModuleNotFoundError

Reply:
I gave it a shot 376dc37 . Seems goofier than calling subprocess and from what I can tell unnecessary since it looks like always (or close enough) we test the installed package.

Reply:
`pykern` has to be installed to run it as a command. That's distinct from what is being tested. For development, we use egg links (or easy-install.pth, probably). That's fine.

`pkcli_test.py` does this work already so not sure why you are duplicating it in another test. Just add argh_test.py to pkcli_data and add a case function.

For "modern" CI and development, it makes sense to have it be installed so we don't have to muck with paths everywhere, e.g. to run pykern. However, there's a principle here regarding testing. Specifically, argh_test.py is data, and we should avoid adding it as an installed pkcli. This mucks up autocomplete. nersc_test is something different, and has to be installed (because it is for NERSC's CI of our docker images). Sometimes it is difficult to separate unit test data, but most times it isn't.

You can verify pkcli_test.py tests the local file with:

```sh
# this should not be found
grep pykern ~/.pyenv/versions/py3/lib/python3.9/site-packages/easy-install.pth
rm -rf ~/.pyenv/versions/py3/lib/python3.9/site-packages/pykern*
# should get not found
pykern
cd ~/src/radiasoft/pykern/tests
# should pass
PYTHONPATH=~/src/radiasoft/pykern/tests pykern test pkcli_test.py
```

Reply:
The commands you gave don't work for me:
```
$ grep pykern ~/.pyenv/versions/py3/lib/python3.9/site-packages/easy-install.pth
/home/vagrant/src/radiasoft/pykern
$ pip uninstall pykern
<snip>
  Successfully uninstalled pykern-20231024.210332
$ grep pykern ~/.pyenv/versions/py3/lib/python3.9/site-packages/easy-install.pth
$ rm -rf ~/.pyenv/versions/py3/lib/python3.9/site-packages/pykern*
$ pykern
bash: pykern: command not found
$ cd ~/src/radiasoft/pykern/tests
tests$ PYTHONPATH=~/src/radiasoft/pykern/tests pykern test pkcli_test.py
bash: pykern: command not found
```

Regardless I made the changes 1afbd34

Reply:
he he. You didn't follow the instructions to the letter. Since it's pyenv, I didn't include `pip uninstall pykern`. This removes `pykern` the command, which is a proxy in pyenv. This should work after a pip uninstall:

`PYTHONPATH=~/src/radiasoft/pykern/tests python ~/src/radiasoft/pykern/pykern/pykern_console.py test pkcli_test.py`

Reply:
They didn't give enough detail so I went rogue. How do I achieve "# this should not be found" w/o an uninstall? Remove it from the file by hand? Something else?

I tried removing it by hand. Now I get 
```
tests (418-argh-pin)$ PYTHONPATH=~/src/radiasoft/pykern/tests pykern test pkcli_test.py
Traceback (most recent call last):
  File "/home/vagrant/.pyenv/versions/py3/bin/pykern", line 33, in <module>
    sys.exit(load_entry_point('pykern', 'console_scripts', 'pykern')())
  File "/home/vagrant/.pyenv/versions/py3/bin/pykern", line 22, in importlib_load_entry_point
    for entry_point in distribution(dist_name).entry_points
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/importlib/metadata.py", line 542, in distribution
    return Distribution.from_name(distribution_name)
  File "/home/vagrant/.pyenv/versions/3.9.15/lib/python3.9/importlib/metadata.py", line 196, in from_name
    raise PackageNotFoundError(name)
importlib.metadata.PackageNotFoundError: pykern
```

Reply:
I ran just the command you gave me 
```
PYTHONPATH=~/src/radiasoft/pykern/tests python ~/src/radiasoft/pykern/pykern/pykern_console.py test pkcli_test.py case=argh
Traceback (most recent call last):
  File "/home/vagrant/src/radiasoft/pykern/pykern/pykern_console.py", line 13, in <module>
    from pykern import pkcli
ModuleNotFoundError: No module named 'pykern'
```

Reply:
Yes, that's it. `pykern` (pykern_console.py) the executable is still in your path, but it is just a simple program that loads pykern pkcli, which does all the work.

Reply:
Yes, I meant to resolve. If you go back to https://github.com/radiasoft/pykern/pull/422#discussion_r1417742112, and do those steps, it will verify that there's no module installed (ModuleNotFoundError) until you add a PYTHONPATH, which will point to the uninstalled directory.